extensions/obviousAlexC/penPlus.js

```diff
@@ -0,0 +1,2621 @@
+          0,
+          1,
+
+          x2,
+          -y2,
+          0,
+          1,
+
+          x3,
+          -y3,
+          0,
+          1,
+        ]);
+      }
+
+      //? Bind Positional Data
+      gl.bindBuffer(gl.ARRAY_BUFFER, depthVertexBuffer);
+      gl.bufferData(gl.ARRAY_BUFFER, vertexBufferData, gl.DYNAMIC_DRAW);
+
+      gl.vertexAttribPointer(
+        a_position_Location_depth,
+        4,
+        gl.FLOAT,
+        false,
+        f32_4,
+        0
+      );
+
+      gl.useProgram(penPlusShaders.depth.ProgramInf.program);
+
+      gl.drawArrays(gl.TRIANGLES, 0, 3);
+
+      gl.bindFramebuffer(gl.FRAMEBUFFER, lastFB);
+    },
+
+    setValueAccordingToCaseTriangle: (
+      targetId,
+      attribute,
+      value,
+      wholeTri,
+      offset
+    ) => {
+      offset = offset + attribute || attribute;
+      let valuetoSet = 0;
+      switch (attribute) {
+        //U
+        case 0:
+          valuetoSet = value;
+          break;
+        //V
+        case 1:
+          valuetoSet = value;
+          break;
+
+        //100 since that is what scratch users are accustomed to.
+        //R
+        case 2:
+          valuetoSet = Math.min(Math.max(value, 0), 100) * 0.01;
+          break;
+        //G
+        case 3:
+          valuetoSet = Math.min(Math.max(value, 0), 100) * 0.01;
+          break;
+        //B
+        case 4:
+          valuetoSet = Math.min(Math.max(value, 0), 100) * 0.01;
+          break;
+
+        //Clamp to 0 so we can't go behind the stage.
+        //Z
+        case 5:
+          if (value < 0) {
+            valuetoSet = 0;
+            break;
+          }
+          valuetoSet = Math.min(value / penPlusAdvancedSettings._maxDepth, 1);
+          break;
+
+        //Clamp to 1 so we don't accidentally clip.
+        //W
+        case 6:
+          if (penPlusAdvancedSettings.wValueUnderFlow == true) {
+            valuetoSet = value;
+          } else {
+            valuetoSet = Math.max(value, 1);
+          }
+          break;
+        //Transparency
+        //Same story as color
+        case 7:
+          valuetoSet = Math.min(Math.max(value, 0), 1000) * 0.01;
+          break;
+
+        //Just break if value isn't valid
+        default:
+          break;
+      }
+      //Check if the index even exists.
+      if (attribute >= 0 && attribute <= 7) {
+        if (wholeTri) {
+          triangleAttributesOfAllSprites[targetId][attribute] = valuetoSet;
+          triangleAttributesOfAllSprites[targetId][attribute + 8] = valuetoSet;
+          triangleAttributesOfAllSprites[targetId][attribute + 16] = valuetoSet;
+        } else {
+          triangleAttributesOfAllSprites[targetId][offset] = valuetoSet;
+        }
+      }
+    },
+  };
+
+  const lilPenDabble = (InativeSize, curTarget, util) => {
+    checkForPen(util);
+
+    const attrib = curTarget["_customState"]["Scratch.pen"].penAttributes;
+
+    Scratch.vm.renderer.penLine(
+      Scratch.vm.renderer._penSkinId,
```

Comment:
Are you sure it's `_penSkinId`? VSCode is complaining about it all over the place.

Reply:
That is the property that give the skin ID for the pen in Turbowarp. I'm sure because every time I've checked it returns the pen skin and also lets me draw to said skin.